pkg/controller/scheduledscaling/scheduled_scaling.go

```diff
@@ -183,30 +191,208 @@ func (c *Controller) runOnce(ctx context.Context) error {
 	return nil
 }
 
-func (c *Controller) scheduleActive(spec v1.ScalingScheduleSpec) (bool, error) {
+func (c *Controller) runOnce(ctx context.Context) error {
+	schedulesInterface := c.scalingScheduleStore.List()
+	schedules := make([]*v1.ScalingSchedule, 0, len(schedulesInterface))
+	for _, scheduleInterface := range schedulesInterface {
+		schedule, ok := scheduleInterface.(*v1.ScalingSchedule)
+		if !ok {
+			return ErrNotScalingScheduleFound
+		}
+		schedules = append(schedules, schedule)
+	}
+
+	clusterschedulesInterface := c.clusterScalingScheduleStore.List()
+	clusterschedules := make([]*v1.ClusterScalingSchedule, 0, len(clusterschedulesInterface))
+	for _, scheduleInterface := range clusterschedulesInterface {
+		schedule, ok := scheduleInterface.(*v1.ClusterScalingSchedule)
+		if !ok {
+			return ErrNotScalingScheduleFound
+		}
+		clusterschedules = append(clusterschedules, schedule)
+	}
+
+	err := c.updateStatus(ctx, schedules, clusterschedules)
+	if err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	log.Info("Adjusting scaling")
+	// TODO: event recorder
+	err = c.adjustScaling(ctx, schedules, clusterschedules)
+	if err != nil {
+		return fmt.Errorf("failed to adjust scaling: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Controller) adjustScaling(ctx context.Context, schedules []*v1.ScalingSchedule, clusterschedules []*v1.ClusterScalingSchedule) error {
```

Comment:
It would really help to split this in 2-3 functions to make it easier to review it.

Reply:
I have split it up, hope it helps.

Reply:
Thanks! I'll give it a read again 🙏